### PR TITLE
Enable more import linters (gci, importas)

### PR DESCRIPTION
**Pull request by sunjayBhatia**
Enables more linters for consistent import ordering/aliasing

importas configuration can be expanded upon, right now this just makes contour api and gateway api imports consistent, needs rules for k8s.io/api/..., internal contour packages, etc.

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -62,6 +64,12 @@ linters-settings:
     enable-all: true
   ginkgolinter:
     forbid-focus-container: true
+  importas:
+    alias:
+    - pkg: github.com/projectcontour/contour/apis/projectcontour/(v\w+)
+      alias: contour_$1
```
Comment (sunjayBhatia) [inline .golangci.yml:70]: considered `projectcontour` here too, but this is shorter and we already use variants of this in many places

File: .golangci.yml [hunk 2]
```diff
@@ -6,6 +6,8 @@ linters:
   - bodyclose
   - gofmt
   - goimports
+  - gci
```
Comment (skriss) [inline .golangci.yml:9]: WDYT about enforcing import grouping? Something like stdlib, third-party, and github.com/projectcontour for groups?

File: .golangci.yml (hunk 1 above)
Comment (skriss) [reply .golangci.yml:70]: SGTM

File: .golangci.yml (hunk 2 above)
Comment (skriss) [reply .golangci.yml:9]: (ref. https://golangci-lint.run/usage/linters/#gci)

Comment (sunjayBhatia) [reply .golangci.yml:9]: yeah I'm down with that 👍🏽 

I guess we should add to the possible new `make format` to format imports with `gci` as well

File: test/conformance/gatewayapi/gateway_conformance_test.go [hunk 3]
```diff
@@ -31,7 +31,7 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/client/config"
 	"sigs.k8s.io/controller-runtime/pkg/log"
-	v1 "sigs.k8s.io/gateway-api/apis/v1"
+	gatewayapi_v1 "sigs.k8s.io/gateway-api/apis/v1"
 	"sigs.k8s.io/gateway-api/apis/v1alpha2"
 	"sigs.k8s.io/gateway-api/apis/v1beta1"
```
Comment (skriss) [inline test/conformance/gatewayapi/gateway_conformance_test.go:36]: Hmm, why aren't these being updated? Does it not add import aliases if they don't already exist?

File: .golangci.yml [hunk 4]
```diff
@@ -62,6 +64,18 @@ linters-settings:
...
   gofumpt:
     extra-rules: true
+  gci:
+    custom-order: true
+    sections:
+    - standard
+    - default
+    - prefix(github.com/projectcontour/contour)
+  importas:
+    alias:
```
Comment (skriss) [inline .golangci.yml:74]: Yeah, we should try to get the core k8s APIs added here as well for consistency, but fine to do that as a follow-up.

File: test/conformance/gatewayapi/gateway_conformance_test.go (hunk 3 above)
Comment (sunjayBhatia) [reply test/conformance/gatewayapi/gateway_conformance_test.go:36]: Yeah i thought these would have changed. There are some more settings in the `importas` linter that might be usable to force things, will take a look

**Comment by sunjayBhatia** [comment]
> I'm good with these rules, sampled the code changes and they look reasonable, but given this is going to cause major merge conflicts for existing PRs we might want to hold off on merging until after the upcoming release, so we don't block any in-flight PRs for this release?

yep sounds good to me, this one touches almost every file i think 😂

**Comment by sunjayBhatia** [comment]
added do not merge label so this is not merged until after the upcoming release

**Review by skriss — APPROVED** [review]
😬 definitely gonna cause some merge conflicts for in-flight work but no better time than right after a release.

**Pull request merged**